dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/domain/FilePersistenceStrategy.kt

```diff
@@ -16,21 +16,17 @@ import java.io.File
 internal open class FilePersistenceStrategy<T : Any>(
     dataDirectory: File,
     serializer: Serializer<T>,
-    recentDelayMs: Long = MAX_DELAY_BETWEEN_MESSAGES_MS,
-    maxBatchSize: Long = MAX_BATCH_SIZE,
-    maxItemsPerBatch: Int = MAX_ITEMS_PER_BATCH,
-    oldFileThreshold: Long = OLD_FILE_THRESHOLD,
-    maxDiskSpace: Long = MAX_DISK_SPACE,
+    filePersistenceConfig: FilePersistenceConfig = FilePersistenceConfig(),
     payloadDecoration: PayloadDecoration = PayloadDecoration.JSON_ARRAY_DECORATION
 ) : PersistenceStrategy<T> {
 
     private val fileOrchestrator = FileOrchestrator(
         rootDirectory = dataDirectory,
-        recentDelayMs = recentDelayMs,
-        maxBatchSize = maxBatchSize,
-        maxLogPerBatch = maxItemsPerBatch,
-        oldFileThreshold = oldFileThreshold,
-        maxDiskSpace = maxDiskSpace
+        recentDelayMs = filePersistenceConfig.recentDelayMs,
```

Comment:
Instead of copy pasting all the params here, you could actually pass in the `filePersistenceConfig` object

Reply:
That was my idea also in the first place but I would rather keep that FileOrchestrator with primitive parameters as we might use it somewhere else. The FilePersistenceConfig it is strictly related with the domain/FilePersistenceStrategy whereas the FileOrchestrator is living somewhere else and it is intended for more general usage. I was not sure that making it dependable of the `FilePersistenceConfig` would be a wise decision but let me know what do you think having all these in mind.

Reply:
IMHO it's not an issue if we make it depend on `FilePersistenceConfig` which is just a data class. It will avoid copying params around.